### ISSUE TITLE
GDPR: Implement UI for opt-in checkbox on signup form

### ISSUE THREAD

**Issue by lyzadanger** [hypothesis/product-backlog#611]
This is part of https://github.com/hypothesis/product-backlog/issues/591

The UI layer of the opt-in checkbox can be implemented separately from the model/logic changes underneath. This issue is to track the UI work for placing the opt-in checkbox into the user signup form.

![image](https://user-images.githubusercontent.com/439947/39128171-16eeba8a-46d5-11e8-936f-cceea9faed06.png)

Whoever picks this up should be familiar with all of the AC in https://github.com/hypothesis/product-backlog/issues/591 as not all details will be repeated here.

## Acceptance Criteria

* [x] Checkbox is visible on user sign-up form with wording and links as shown in comps
* [x] Checkbox should be unchecked by default
* [x] If form is submitted without checkbox checked, the form should error, no user should be created
* [x] Form error (tooltip) should include text "Acceptance of the privacy policy is required"

## Discussion

While the checkbox will technically tie into a "real" field in the database, it isn't necessary to match the naming of the checkbox field in the form to the database field (unless you want to) as the database field is a different format, anyway (it'll store a datetime).

## Out of Scope

* Any sort of logic for "doing anything" to the user data during user creation—this is a UI-level concern only

## Important Note

Given that the result of this work will be a checkbox that does nothing, we should not merge this work into `master` until the rest of the pieces are ready—we'll need to coordinate.

**Comment by robertknight** [comment]
This is currently being deployed.
